src/KlisterPath.hs

```diff
@@ -0,0 +1,91 @@
+{-
+Module           : KlisterPath
+Description      : Implementation of $KLISTERPATH
+Copyright        : (c) Klister Collaborators 2020
+License          : See GH#55
+Maintainer       : Langston Barrett <langston@galois.com>
+Stability        : provisional
+-}
+
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+
+module KlisterPath
+  ( KlisterPath
+  , KlisterPathError(..)
+  , ppKlisterPathError
+  , getKlisterPath
+  , makeKlisterPath
+  , importFromKlisterPath
+  ) where
+
+import Data.Functor ((<&>))
+import qualified Data.Set as Set
+import Data.Set (Set)
+import qualified Data.Text.Prettyprint.Doc as PP
+import Data.Traversable (for)
+import System.Directory (canonicalizePath, listDirectory)
+import System.FilePath.Posix ((</>))
+import System.Environment (lookupEnv)
+
+-- | A list of directories to search for modules
+--
+-- Invariant: All paths are canonical.
+newtype KlisterPath = KlisterPath { unKlisterPath :: Set FilePath }
+  deriving (Monoid, Semigroup, Show)
+
+makeKlisterPath :: Set FilePath -> IO KlisterPath
+makeKlisterPath paths =
+  KlisterPath . Set.fromList <$>
+    mapM canonicalizePath (Set.toList paths)
+
+-- | Fetch the Klister path from the environment variable @KLISTERPATH@.
+getKlisterPath :: IO KlisterPath
+getKlisterPath =
+  makeKlisterPath =<<
+    maybe Set.empty (Set.fromList . (split ':')) <$>
+      lookupEnv "KLISTERPATH"
+  where
+    -- How is this not in the Prelude...?
+    split :: Eq a => a -> [a] -> [[a]]
```

Comment:
It seems to me that this will fail on directory names that contain a colon. OTOH, I don't think POSIX `PATH` works right in that way either, and AFAIK Windows doesn't even allow colons in directory names, so we're probably fine :-)

Reply:
Yeah, I don't think there's much to be gained by handling edge cases here :-)